resources/js/components/GraphqlMutation.vue

```diff
@@ -84,6 +89,12 @@ export default {
     created() {
         this.initialVariables = JSON.parse(JSON.stringify(this.variables))
         this.data = this.variables
+
+        if (this.debounce) {
+            this.mutate = useDebounceFn(this.mutateFn, this.debounce)
```

Comment:
Doe this method also return a promise that only resolves once mutateFn is resolved?

Otherwise this may cause issues with functions wanting to await or chain onto the mutate function e.g. https://github.com/rapidez/core/blob/97d2b62daf7953c313aa7d59e023e174ef3775fe/resources/views/checkout/steps/shipping_address.blade.php#L19

Reply:
Yes, `useDebounceFn` actually returns a promise by default (see docs https://vueuse.org/shared/useDebounceFn/) and it looks to also properly wait for the async function inside to finish running before resolving.

There does seem to be some weirdness with it also resolving continuously while you're spamming the button. This may be a vueUse bug because I don't really see why this would happen. I don't know if this will affect any current use cases though.